disk-mapper/internal/setup/setup.go

```diff
@@ -165,6 +167,49 @@ func (s *Manager) saveConfiguration(passphrase []byte) error {
 	return s.config.Generate(stateDiskMappedName, s.diskPath, filepath.Join(keyPath, keyFile), cryptsetupOptions)
 }
 
+// LogDevices logs all available block devices and partitions (lsblk like).
+func (s *Manager) LogDevices() {
+	var devices []fs.FileInfo
+	dirs, err := os.ReadDir("/sys/class/block")
+	for _, file := range dirs {
+		if file.IsDir() {
+			continue
+		}
+		fileInfo, err := file.Info()
+		if err != nil {
+			panic(err)
+		}
+		devices = append(devices, fileInfo)
+	}
+	if err != nil {
+		panic(err)
+	}
+
+	s.log.Infof("List of all available block devices and partitions:")
+	for _, device := range devices {
+		var stat syscall.Statfs_t
+		dev := "/dev/" + device.Name()
+		if err := syscall.Statfs(dev, &stat); err != nil {
+			s.log.Errorf("failed to statfs %s: %v", dev, err)
+		}
```

Comment:
```suggestion
		if err := syscall.Statfs(dev, &stat); err != nil {
			s.log.With(zap.Error(err)).Errorf("failed to statfs %s", dev )
		}
```
Is it OK to continue if statfs failed? Or should we continue with the next device?

Reply:
Since there are several reasons why this could fail (invalid permissions, signal interrupt, I/O error, ...) I think we should not fail but instead continue with the next device.